src/cache/index.ts

```diff
@@ -27,7 +28,17 @@ export async function cachedApiCall(
 
   if (response.statusCode == null || response.statusCode >= 400) {
     const error = response.body as any
-    throw Error(error.message as string)
+    switch (response.statusCode) {
+      case 400:
+        if (error.message === 'Your API token is invalid.') {
```

Comment:
This is very fragile but I guess there's no other way to identify this error right?

Reply:
I checked [football-data's API documentation on errors](https://docs.football-data.org/general/v4/errors.html) but it's very vague. Error responses will have an `error` property with a readable string, like this:

```json
{
   "error": "Argument 'id' is expected to be an integer in a specific range."
}
```

No unique id or error codes, not even a list of possible strings. So I guess we're stuck with relying on known strings found out empirically.